runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/DoFnOp.java

```diff
@@ -217,7 +222,9 @@ public void open(
               .stateInternalsForKey(null)
               .state(StateNamespaces.global(), StateTags.bag(bundleStateId, windowedValueCoder));
       final ExecutableStage executableStage = ExecutableStage.fromPayload(stagePayload);
-      stageBundleFactory = samzaExecutionContext.getJobBundleFactory().forStage(executableStage);
+      final ExecutableStageContext stageContext =
+          SamzaExecutableStageContextFactory.getInstance().get(jobInfo);
+      stageBundleFactory = stageContext.getStageBundleFactory(executableStage);
```

Comment:
According to the deprecated notes
```
 * @deprecated replace with a {@link DefaultJobBundleFactory} when appropriate if the {@link
 *     EnvironmentFactory} is a {@link
 *     org.apache.beam.runners.fnexecution.environment.DockerEnvironmentFactory}, or create an
 *     {@code InProcessJobBundleFactory} and inline the creation of the environment if appropriate.
 */
 ```
 I don't think the environment is docker always. Does this still work for other environments?

Reply:
Actually Yes, DefaultJobBundleFactory supports Docker, Process, External and Embeded Environment

You may check https://github.com/apache/beam/blob/243128a8fc52798e1b58b0cf1a271d95ee7aa241/runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DefaultJobBundleFactory.java#L118